src/Spec.ts

```diff
@@ -116,46 +116,79 @@ export type Warning =
 function wrapWarn(source: string, dom: any, warn: (err: EcmarkupError) => void) {
   return (e: Warning) => {
     let { message, ruleId } = e;
-    let line: number | undefined, column: number | undefined;
+    let line: number | undefined;
+    let column: number | undefined;
+    let nodeType: string;
     if (e.type === 'global') {
       line = undefined;
       column = undefined;
+      nodeType = 'html';
     } else if (e.type === 'raw') {
       ({ line, column } = e);
+      nodeType = 'html';
     } else {
-      let nodeLoc = utils.getLocation(dom, e.node);
       if (e.type === 'node') {
-        ({ line, col: column } = nodeLoc.startTag);
+        if (e.node.nodeType === 3) {
```

Comment:
```suggestion
        if (e.node.nodeType === Node.TEXT_NODE) {
```

Reply:
That's not a global in Node, but I'll add a comment.